bundles/framework/announcements/view/AnnouncementsCollapse.jsx

```diff
@@ -0,0 +1,70 @@
+import React from "react";
+import PropTypes from 'prop-types';
+import { Controller, LocaleConsumer } from 'oskari-ui/util';
+import { Message } from 'oskari-ui';
+import { AnnouncementsModal } from './AnnouncementsModal';
+import { Collapse, Divider } from 'antd';
+
+//Collapse panel -> set title, content and date range according to announcement
+
+const { Panel } = Collapse;
+
+const AnnouncementsCollapse = ({controller, updated, panels, modals, checked }) => {
+
+      if (!updated) {
+        controller.getAnnouncements(function(ann){
+
+        
+          var panels = [];
+          var modals = [];
+            ann.data.forEach((announcement) => {
+
+              if (announcement.active) {
+                //if announcement is active, then show pop-up of the content
+                modals.push(announcement);
+              }
+              panels.push(
+                      <Panel header={announcement.title} key={announcement.id}>
+                          <h3>{announcement.title}</h3>
+                          <p>{announcement.content}</p>
+                          <Divider />
+                          <b><Message messageKey={'valid'} /></b>
+                          <p>{announcement.begin_date} - {announcement.end_date}</p>
+                      </Panel>
+                );
+            });
+            controller.updatePanelsModals(panels,modals);
+          });
+      }
```

Comment:
What's happening here? What does "updated" mean? Instead of updated variable why not just pass the announcements and create the panels on render based on that? This seems weirdly complex way of triggering loading of announcements if that is what this is.

Reply:
Could not figure out how to do it without creating an error. The render happened every time before the announcements were fetched so there was no content and it lead to an error. Suggestions on how to render only after the announcements are fetched?